src/persistence/test/MCWSPersistenceProviderSpec.js

```diff
@@ -1,189 +1,186 @@
+            },
+            {
+                Subject: '/some/namespace/123-ABC',
+                Predicate: 'has MIO type',
+                Object: 'opaque_file'
+            }
+        ];
+        mockNamespace.read.and.returnValue(Promise.resolve(namespaceTriples));
+        
+        const objects = await mcwsPersistenceProvider.getNamespacesFromMCWS({ url: '/test/namespace' });
+        expect(objects).toEqual(namespaceTriples);
+        done();
+    });
 
-        it("allows objects to be created", function (done) {
-            mockOpaqueFile.create.and.returnValue(Promise.resolve(true));
-            persistence
-                .createObject(
-                    "testSpace",
-                    "testKey",
-                    { someKey: "some value" }
-                )
-                .then(function (result) {
-                    expect(result).toBe(true);
-                    expect(mockNamespace.opaqueFile)
-                        .toHaveBeenCalledWith("testKey");
-                    expect(mockOpaqueFile.create)
-                        .toHaveBeenCalledWith({ someKey: "some value" });
-                    done();
-                });
-        });
+    it('allows objects to be created', async (done) => {
+        mockOpaqueFile.create.and.returnValue(Promise.resolve(true));
+        const domainObject = {
+            identifier: {
+                key: 'testKey',
+                namespace: 'testSpace'
+            },
+            someKey: 'some value'
+        };
+        const result = await mcwsPersistenceProvider.create(domainObject);
+        
+        expect(result).toBe(true);
+        expect(mockNamespace.opaqueFile)
+            .toHaveBeenCalledWith('testKey');
+        expect(mockOpaqueFile.create)
+            .toHaveBeenCalledWith({ someKey: 'some value' });
+        done();
+    });
 
-        it("allows objects to be read", function (done) {
-            mockOpaqueFile.read.and.returnValue(Promise.resolve("test object"));
-            persistence
-                .readObject("testSpace", "testKey")
-                .then(function (object) {
-                    expect(object).toBe("test object");
-                    expect(mcws.namespace)
-                        .toHaveBeenCalledWith("/test/namespace");
-                    expect(mockNamespace.opaqueFile)
-                        .toHaveBeenCalledWith("testKey");
-                    expect(mockOpaqueFile.read)
-                        .toHaveBeenCalled();
-                    done();
-                });
-        });
+    it('allows objects to be read', async (done) => {
+        const identifier = {
+            key: 'testKey',
+            namespace: 'testSpace'
+        };
+        const model = {
+            someKey: 'some value'
+        };
+        const domainObject = {
+            identifier,
+            ...model
+        };
+        mockOpaqueFile.read.and.returnValue(Promise.resolve({json: () => Promise.resolve(model)}));
```

Comment:
i don't think we have models anymore without legacy domain objects?

Reply:
We basically store only models in MCWS, we strip out the identifier (similar to how do it in couchdb). So, when we retrieve the item from MCWS we use the key to get the model and we know the namespace, so we build the identifier in the provider before returning it to open mct.